/*
 * Copyright (c) 2021 NVIDIA CORPORATION & AFFILIATES, ALL RIGHTS RESERVED.
 *
 * This software product is a proprietary product of NVIDIA CORPORATION &
 * AFFILIATES (the "Company") and all right, title, and interest in and to the
 * software product, including all associated intellectual property rights, are
 * and shall remain exclusively with the Company.
 *
 * This software product is governed by the End User License Agreement
 * provided with the software product.
 *
 */

syntax = "proto3";

import "common.proto";

/* Host (x86) -> DPU (Arm):
 * ========================
 * gRPC server that exposes the application's API to the host
 * for remote management.
 */
service URLFilter {
	/* Subscribe to receive log events from the DPU */
	rpc Subscribe (SubscribeReq) returns (stream LogRecord);

	/* Create a new rule work-set, possibly resetting the current work-set */
	rpc Create (CreateReq) returns (CreateResp);

	/* Add a new URL Filter rule */
	rpc Add (FilterRule) returns (AddResp);

	/* Compile the work-set and commit it to the DPI engine */
	rpc Commit (CommitReq) returns (CommitResp);

	/* Quit the App */
	rpc Quit (QuitReq) returns (QuitResp);
}

/* Here for future compatibility */
message SubscribeReq {
}

/* Here for future compatibility */
message CreateReq {
}

/* Here for future compatibility */
message CreateResp {
}

message FilterRule {
	string msg = 1;
	string pcre = 2;
}

/* Here for future compatibility */
message AddResp {
}

/* Here for future compatibility */
message CommitReq {
}

/* Here for future compatibility */
message CommitResp {
}

/* Here for future compatibility */
message QuitReq {
}

/* Here for future compatibility */
message QuitResp {
}
